app/request.go

```diff
@@ -63,8 +86,90 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
 				},
 			},
 			Action: func(ctx *cli.Context) error {
-				return c.getRequestStatus(ctx.String(NamespaceFlagName), ctx.String("request-id"))
+				res, err := c.getRequestStatus(ctx, ctx.String("request-id"))
+				if err != nil {
+					return err
+				}
+				return PrintProto(res)
+			},
+		}, {
+			Name:    "wait",
+			Usage:   "wait till the request completes",
+			Aliases: []string{"w"},
+			Flags: []cli.Flag{
+				NamespaceFlag,
+				&cli.StringFlag{
+					Name:     "request-id",
+					Usage:    "the request-id of the asynchronous request",
+					Aliases:  []string{"r"},
+					Required: true,
+				},
+			},
+			Action: func(ctx *cli.Context) error {
+				return c.waitOnRequest(ctx, "", ctx.String("request-id"))
 			},
 		}},
 	}}, nil
 }
+
+func (c *RequestClient) waitOnRequest(ctx *cli.Context, operation string, requestID string) error {
+
+	ticker := time.NewTicker(time.Millisecond)
+	timer := time.NewTimer(ctx.Duration(RequestTimeoutFlagName))
+
+	writer := uilive.New()
+	writer.Start()
+	defer writer.Stop()
+
+loop:
+	for {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("timed out waiting for request, namespace=%s, requestID=%s, timeout=%s",
+				ctx.String(NamespaceFlagName),
+				requestID,
+				ctx.Duration(RequestTimeoutFlagName),
+			)
+		case <-ticker.C:
+			status, err := c.getRequestStatus(ctx, requestID)
+			if err != nil {
+				return err
+			}
+			switch status.State {
+			case request.STATE_FULFILLED:
+				break loop
+			case request.STATE_FAILED:
+				fmt.Fprintf(writer, "operation failed \n")
+				return fmt.Errorf("request failed: %s", status.FailureReason)
+			case request.STATE_CANCELLED:
+				fmt.Fprintf(writer, "operation failed \n")
+				return fmt.Errorf("request was cancelled: %s", status.FailureReason)
+			}
+			if operation != "" {
+				fmt.Fprintf(writer, "waiting for %s operation to finish, current state: %s\n",
+					operation, request.RequestStatus_State_name[int32(status.State)])
+			} else {
+				fmt.Fprintf(writer, "waiting for request with '%s' id to finish, current state: %s\n",
+					requestID, request.RequestStatus_State_name[int32(status.State)])
+			}
+			ticker.Reset(time.Second * time.Duration(status.CheckDuration.Seconds))
+		}
+	}
+	if operation != "" {
+		fmt.Fprintf(writer, "%s operation completed successfully\n", operation)
+	} else {
+		fmt.Fprintf(writer, "request with '%s' id to finished successfully\n", requestID)
+	}
+	return nil
+}
+
+func (c *RequestClient) HandleRequestStatus(
+	ctx *cli.Context,
+	operation string,
+	status *request.RequestStatus,
+) error {
+	if ctx.Bool(AsyncRequestFlagName) {
+		return PrintProto(status)
+	}
```

Comment:
please print out the request object so if someone ctrl-Cs and wants to check after the fact, they are able to do so